first-aid.lic

```diff
@@ -8,24 +8,46 @@ class FirstAid
   include DRC
 
   def initialize
+    settings = get_settings
     @chart_data = get_data('anatomy-charts').first_aid_charts
-
-    train_first_aid
+	@textbook = settings.textbook
+    if @textbook
+      @booktype = 'textbook'
+	  textbook_charts
+    else
+      @booktype = 'compendium'
+	  compendium_charts
+	end
   end
-
-  def train_first_aid
+  
+  def compendium_charts
     bput('get my compendium', 'You get')
     bput('look my compendium', '^The compendium lies open to the section on .* physiology')
     pause
     charts = reget(40).grep(/^   .+$/).map(&:strip)
 
     charts_to_read = charts.select { |chart| @chart_data[chart]['scholarship'] <= DRSkill.getrank('Scholarship') }
-
+	study_charts(charts_to_read)
+  end
+  
+  def textbook_charts
+    charts_to_read = []
+    bput('get my textbook', 'You get')
+    readable_charts = @chart_data.select { |chart| @chart_data[chart]['scholarship'] <= DRSkill.getrank('Scholarship')}
+    readable_charts.each do |name, info|
```

Comment:
I may be wrong here - but I feel as though I need charts_to_read to be an array of the 'index' (name) in the order that I want them read since that's how study_charts works.  It starts at the first one, studies, turns to the next one.  so on and so forth.

Since I needed to feed that index into an array I used unshift only because the charts in chart_data are already ordered easiest to hardest.  Unshift lets me put them in at the first spot and move everything bag - so it ends ordered hardest to easiest instead.

(just trying to put out why I did it this way - still trying to find a way to change it)